test/run_test.py

```diff
@@ -340,6 +349,82 @@ def print_to_stderr(message):
     print(message, file=sys.stderr)
 
 
+def get_test_time_reports_from_S3():
```

Comment:
Nit (don't forget to import `Any`) 
```suggestion
def get_test_time_reports_from_S3() -> List[Dict[str, Any]]:
```